app/models/user.rb

```diff
@@ -3,5 +3,11 @@ class User < ApplicationRecord
 
   has_secure_password
 
+  has_many :gift_card_orders, dependent: :destroy
+
   validates :username, presence: true, uniqueness: true
+
+  def balance(currency = Dollar, account_type = AccountType::User::PRIMARY)
+    account(account_type, currency).balance.cents
```

Comment:
Why cents? Seems like you'd have to divide by 100 when you need to know the USD balance?